lib/basecrm/services/contacts_service.rb

```diff
@@ -103,6 +103,28 @@ def update(contact)
       Contact.new(root[:data])
     end
 
+    # Upsert a contact
+    #
+    # post '/contacts/upsert?filter_param=filter_value 
+    #
+    # Create a new contact or update an existing, based on a value of a filter or a set of filters. 
+    # At least a single filter - query parameter - is required. If no parameters are present, the request will return an error.
+    # See full docs https://developers.getbase.com/docs/rest/reference/contacts
+    #
+    # @param filters [Hash] - hash contain filters, one level deep e.g. { name: 'string', 'custom_fields[field]': 'value' }
+    # @param contact [Contact, Hash] - This object's attributes describe the object to be updated or created
+    # @return [Contact] The resulting object representing updated or created resource.
+    def upsert(filters, deal)
+      validate_upsert_filters!(filters)
+      validate_type!(deal)
```

Comment:
`validate_type!` will fail if the argument is not a Contact or Hash.

```suggestion
      validate_type!(contact)
```